### PR TITLE
Add: Consistent page headings to improve wayfinding

### PR DESCRIPTION
**Pull request by danhauk**
#### Changes proposed in this Pull Request

* This PR adds page headings to screens across all of Calypso, using the existing `<FormattedHeader>` component.
* It also refactors the `<SidebarNavigation>` component (the component on mobile viewports that opens the sidebar navigation) to clarify the functionality.

#### Background
There are many overarching issues with the information architecture (IA) of Calypso. Ultimately, these will need to be fixed to provide a great experience to all of our users. When the overall IA is improved it will help reduce the cognitive load of finding your way around.

Until that happens, however, adding a consistent page heading pattern throughout Calypso can help bridge the gap.

Some issues this approach solves:

* **Context.** Adding consistent page headings provides context to users about where they are in relation to the rest of Calypso. Currently the only real context provided is seeing the section highlighted in the sidebar nav. On mobile, the sidebar is hidden, so this context is non-existent.
* **Accessibility.** When using assistive technology like a screen reader, the page does not have a clear title.

More background about design exploration, proposals, and why this specific development approach was chosen can be found at p9Jlb4-12y-p2

#### Screenshots

Mobile Before | Mobile After
------------ | -------------
Screen Shot 2019-11-07 at 11 44 33 AM | Screen Shot 2019-11-07 at 11 44 07 AM

Desktop Before | Desktop After
---- | ----
image | image

#### Testing instructions

* Checkout this branch.
* Go through each section of My Site(s) and check for a consistent page heading component.
* Make sure all sections have a page heading, including sub-sections and filters.
* Try to break it.
* Check mobile viewports to make sure it all still works and looks ok.

#### Known issues
* This only adds the page heading to each page. Issues like copy duplication when list headers match the page heading will be addressed in future PRs.
* Sections in /me are not touched with this PR

### CODE REVIEW

**Comment by sixhours** [comment]
Looking good so far! I gave this a spin and found a few issues and have a couple questions, nothing huge:

* The Earn page doesn't have a title; is this intentional?
* Noticed an inconsistency on Import and Export screens. The headings are "Import Your Content" vs. "Export your content". (This one jumped out at me because I clicked Export right after I looked at Import, but worth checking all headings for consistent caps.)
* "WordPress Themes" should probably be "Themes" according to Ran's P2 post. Having WordPress in the title seems redundant.
* This may be out of scope, but some areas (Pages, Domains, etc.) add a "heading" in the action bar thingee (that's the technical term) above the content. Should we remove those duplicates or leave that for a separate PR?

Screen Shot 2019-11-07 at 12 15 48 PM

* I noticed we use `FormattedHeader` a lot across Jetpack screens, but I'm not sure how to test those...something to keep an eye on since the styles for the global component have changed.
* Would it make sense to add an `align` prop to `FormattedHeader` with options of `left`, `right`, and `center` instead of a specific property for left alignment? I don't know if we'd have reason to right-align a heading but it might make the new prop more flexible for future use.

File: client/my-sites/importer/section-import.jsx [hunk 1]
```diff
@@ -299,8 +300,13 @@ class SectionImport extends Component {
 
 		return (
 			<Main>
-				<DocumentHead title={ translate( 'Import' ) } />
+				<DocumentHead title={ translate( 'Import Your Content' ) } />
```
Comment (a8ci18n) [inline client/my-sites/importer/section-import.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Import your content' )` *ES Score: 7*

**Comment by danhauk** [comment]
Thanks for giving it a look @sixhours! I pushed up those copy changes you suggested.

> The Earn page doesn't have a title; is this intentional?

It was, mostly because Earn has that big banner at the top and frankly adding a page heading above that doesn't look good. Maybe we do it anyway to draw more attention to what needs to be fixed with the layout there?

image

> This may be out of scope, but some areas (Pages, Domains, etc.) add a "heading" in the action bar thingee (that's the technical term) above the content. Should we remove those duplicates or leave that for a separate PR?

I was planning to leave those for a separate PR to keep this one focused on just adding page heading elements.

 > I noticed we use FormattedHeader a lot across Jetpack screens, but I'm not sure how to test those...something to keep an eye on since the styles for the global component have changed.

I know we use them a lot in the Jetpack Connect flow. I gave that a look to make sure the global styles I changed didn't affect existing instances. Worth a second set of eyes though. To do that:

* Spin up a new Jetpack site on jurassic.ninja
* Click to connect Jetpack on wp-admin
* When you get to the Jetpack connect flow, replace `wordpress.com` with `calypso.localhost:3000` in the URL and it should work

> Would it make sense to add an align prop to FormattedHeader with options of left, right, and center instead of a specific property for left alignment?

I love this idea, and agree it would make it more flexible for the future.

File: client/my-sites/exporter/section-export.jsx [hunk 2]
```diff
@@ -49,8 +49,9 @@ const SectionExport = ( { isJetpack, canUserExport, site, translate } ) => {
 			<Fragment>
 				<FormattedHeader
 					className="exporter__section-header"
-					headerText={ translate( 'Export your content' ) }
+					headerText={ translate( 'Export Your Content' ) }
```
Comment (a8ci18n) [inline client/my-sites/exporter/section-export.jsx]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'exporting your content' )` *ES Score: 6*

**Comment by sixhours** [comment]
The only thing that jumped out at me is the "Earn" heading is centered:

Screen Shot 2019-11-12 at 10 21 34 AM

And this is not a blocker, but I think we could get away with reducing the font size for smaller screens. 24px feels large. 21px is smaller without losing the sense of hierarchy. What do you think?

Screen Shot 2019-11-12 at 10 27 52 AM

**Comment by sixhours** [comment]
Oh, other than the Earn heading, I say this is 100% 🚢 -able. :D

**Comment by danhauk** [comment]
@sixhours Thanks for giving it another look. I fixed the Earn title alignment and agree with your font size suggestion. I brought it down to 20px to keep with more of a grid-like type scale.

File: client/my-sites/stats/wordads/index.jsx [hunk 3]
```diff
@@ -148,6 +150,11 @@ class WordAds extends Component {
 				/>
 				<PrivacyPolicyBanner />
 				<SidebarNavigation />
+				<FormattedHeader
+					className="wordads__section-header"
+					headerText={ translate( 'Stats and Insights' ) }
+					leftAlign
```
Comment (Aurorum) [inline client/my-sites/stats/wordads/index.jsx]: I think this is a typo, but it's causing a weird jump when switching through Stats tabs. I proposed a fix in #37778. :)

Screenshot 2019-11-20 at 17 05 42

**Pull request merged**